src/ophyd_async/core/signal_backend.py

```diff
@@ -35,6 +35,10 @@ async def get_reading(self) -> Reading:
     async def get_value(self) -> T:
         """The current value"""
 
+    @abstractmethod
+    async def get_setpoint(self) -> T:
+        """The current setpoint"""
```

Comment:
without going into bluesky.protocols and looking at the definition of a `Location` protocol and its comments for `setpoint` and `readback`, this docstring is a little confusing. Could we perhaps replace it with:

```suggestion
    async def get_setpoint(self) -> T:
        """The point that a signal was requested to move to."""
```